Dropout.lua

```diff
@@ -20,13 +21,14 @@ function Dropout:updateOutput(input)
       self.output:resizeAs(input):copy(input)
    end
    if self.p > 0 then
-      if self.train then
-         self.noise:resizeAs(input)
-         self.noise:bernoulli(1-self.p)
+      local si = self.si and not self.train
+      if self.train or si and not self.v2 then
```

Comment:
Are you sure about this condition?
Just a few lines downstream in the same `if` block, there is the case where `v2` is used considered, but it will never reach it with the current condition.